packages/main/cypress/specs/DateRangePicker.cy.tsx

```diff
@@ -0,0 +1,570 @@
+				expect(dayInBetween).to.have.attr("aria-selected", "true");
+				expect(endSelectionDay).to.have.attr("aria-selected", "true");
+			});
+	});
+
+	it("Min and max dates are set without format-pattern by using ISO (YYYY-MM-dd) format", () => {
+		cy.wrap({ setLanguage })
+			.invoke("setLanguage", "bg");
+
+		cy.mount(<DateRangePicker minDate="2023-02-10" maxDate="2023-07-22"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.shadow()
+			.find("input")
+			.realClick()
+			.realType("10.02.2023 - 25.07.2023")
+			.realPress("Enter");
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value-state", "Negative");
+	});
+
+	it("Should open month picker if format-pattern is 'MM.yyyy'", () => {
+		cy.mount(<DateRangePicker formatPattern="MM.yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realPress("F4");
+
+		cy.get("@dateRangePicker")
+			.shadow()
+			.find("ui5-calendar")
+			.shadow()
+			.find("ui5-monthpicker")
+			.should("exist")
+			.and("be.visible");
+	});
+
+	it("Select month range in MonthPicker", () => {
+		cy.wrap({ setLanguage })
+			.invoke("setLanguage", "en");
+
+		cy.mount(<DateRangePicker formatPattern="MM.yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("09.2024 - 11.2024")
+			.realPress("Enter")
+			.realPress("F4");
+
+		cy.get<DateRangePicker>("@dateRangePicker")
+			.shadow()
+			.find("ui5-calendar")
+			.shadow()
+			.find("ui5-monthpicker")
+			.shadow()
+			.find(".ui5-mp-root .ui5-mp-item")
+			.then(months => {
+				const startSelectionMonth = months[8];
+				const monthInBetween = months[9];
+				const endSelectionMonth = months[10];
+
+				expect(startSelectionMonth).to.have.class("ui5-mp-item--selected");
+				expect(monthInBetween).to.have.class("ui5-mp-item--selected-between");
+				expect(endSelectionMonth).to.have.class("ui5-mp-item--selected");
+			});
+	});
+
+	it("Should open year picker if format-pattern is 'yyyy'", () => {
+		cy.mount(<DateRangePicker formatPattern="yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realPress("F4");
+
+		cy.get("@dateRangePicker")
+			.shadow()
+			.find("ui5-calendar")
+			.shadow()
+			.find("ui5-yearpicker")
+			.should("exist")
+			.and("be.visible");
+	});
+
+	it("Select year range in YearPicker", () => {
+		cy.wrap({ setLanguage })
+			.invoke("setLanguage", "en");
+
+		cy.mount(<DateRangePicker formatPattern="yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
```

Comment:
i think it will be good to ensure that the input is focused after the click and before we type